### PR TITLE
Use Set instead of setify

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

File: src/fsharp/TastOps.fs [hunk 1]
```diff
@@ -5433,13 +5433,13 @@ let mapTargetsOfDecisionTree f tree = mapAccTipsOfDecisionTree (fun es i -> TDSu
 
 // Dead target elimination 
 let eliminateDeadTargetsFromMatch tree (targets:_[]) =
-    let used = accTargetsOfDecisionTree tree [] |> ListSet.setify (=) |> Array.ofList
-    if used.Length < targets.Length then
-        Array.sortInPlace used;
+    let used = accTargetsOfDecisionTree tree [] |> HashSet
+    if used.Count < targets.Length then
+        let used = used |> Seq.toArray
```
Comment (vasily-kirichenko) [inline src/fsharp/TastOps.fs]: Could we use `Seq.iteri` instead, without converting the set to array?

Comment (forki) [reply src/fsharp/TastOps.fs]: It's used below in array.map again

**Comment by forki** [comment]
done

File: src/fsharp/TastOps.fs [hunk 2]
```diff
@@ -5433,13 +5433,13 @@ let mapTargetsOfDecisionTree f tree = mapAccTipsOfDecisionTree (fun es i -> TDSu
 
 // Dead target elimination 
 let eliminateDeadTargetsFromMatch tree (targets:_[]) =
-    let used = accTargetsOfDecisionTree tree [] |> ListSet.setify (=) |> Array.ofList
-    if used.Length < targets.Length then
-        Array.sortInPlace used;
```
Comment (dsyme) [inline src/fsharp/TastOps.fs]: What's happened to the sorting?

Comment (forki) [reply src/fsharp/TastOps.fs]: reapplied sorting by using a F# set

Comment (dsyme) [reply src/fsharp/TastOps.fs]: I don't get it - why not just keep the sortInPlace?  Converting to a binary tree and back just to sort a freshly allocated array is kind of wasteful.

Comment (forki) [reply src/fsharp/TastOps.fs]: Yeah. This is a bit too much speculation here. Closing for now

**Pull request closed without merging**